hls4ml/model/optimizer/passes/infer_precision.py

```diff
@@ -1,4 +1,5 @@
 import math
+import struct
```

Comment:
Struct is much too low level for what we are doing here. We have a python float. We should use it, not look at the bits.